src/chrome/chromeDebugAdapter.ts

```diff
@@ -1359,19 +1361,25 @@ export abstract class ChromeDebugAdapter implements IDebugAdapter {
         };
     }
 
-    private scriptToScriptEvent(script: Crdp.Debugger.ScriptParsedEvent): ScriptEvent {
+    private scriptToScriptEvent(script: Crdp.Debugger.ScriptParsedEvent): Promise<ScriptEvent> {
         const sourceReference = this.getSourceReferenceForScriptId(script.scriptId);
         const origin = this.getReadonlyOrigin(script.url);
-        const source = {
-            name: path.basename(script.url),
-            path: script.url,
-            sourceReference,
-            origin
-        };
 
-        const clientScript: Script = new Script(source);
+        return new Promise((resolve, reject) => {
```

Comment:
You don't need to create a new Promise, since you're working with `utils.existsAsync` that returns a promise.